### DISCUSSION TITLE
chinese input，fcitx5

### DISCUSSION THREAD

**Discussion by 2061360308** [termux/termux-packages#22380] [Q&A]
I want to use fcitx5 to type Chinese.

environment：termux（no proot-distro），lxqt，fcitx5

step：
```
# 安装必要的包
pkg install fcitx5 fcitx5-configtool fcitx5-chinese-addons fcitx5-gtk-common fcitx5-hangul fcitx5-qt

# 配置环境变量
export XMODIFIERS=@im=fcitx
export GTK_IM_MODULE=fcitx
export QT_IM_MODULE=fcitx
# 启动 D-Bus 服务
termux-services start

eval `dbus-launch --sh-syntax --exit-with-session`

# 启动 fcitx5
fcitx5 &
```

result：
![mmexport1732474799436.jpg](https://github.com/user-attachments/assets/7bebccaf-9c7d-4164-a72b-2f3705ac6687)

If I don't use `eval 'dbus-launch --sh-syntax --exit-with-session'`

![mmexport1732448386340.jpg](https://github.com/user-attachments/assets/b45a7f0d-5275-40b8-ae42-1144ea4cbc47)

**How can I input Chinese without using proot-distro, lxqt and termux-x11 GUI environment?**__

**Comment by cnjhb** [comment]
i don't know

**Comment by cnjhb** [comment]
你可以使用我最新打包的ibus-libpinyin，推荐使用gnome或者xfce这种基于gtk的桌面环境。因为ibus是基于gtk的。
顺带一提这个回复就是我用ibus-libpinyin打得。
<img width="1824" height="976" alt="2026-01-13-222537_1824x976_scrot" src="https://github.com/user-attachments/assets/31751be2-2f51-4cf1-8100-a91c0936db02" />

**Reply by 2061360308** [reply]
👍🏻，马上去试试

**Reply by jerrychan7** [reply]
对于我来说需要多装个gtk3的包

```bash
pkg install ibus ibus-libpinyin libdbusmenu-gtk3
ibus-daemon -drx --panel=/data/data/com.termux/files/usr/libexec/ibus-ui-gtk3
```

**Comment by shiumano** [comment]
If you are willing to use proot for just a specific part of the system, there is a way to run only fcitx5 within proot.

This allows you to use multilingual input in applications like Firefox or the terminal while maintaining native performance.

```bash
ime="fcitx5-mozc"  # Because I am Japanese.
lang_pack="language-pack-ja"
proot-distro install ubuntu
proot-distro login ubuntu -- bash -c \
"apt update && apt upgrade -y && apt install ${ime} fonts-noto-{cjk,color-emoji} ${lang_pack} -y && useradd termux && usermod -o -u $(id -u) termux"

lang="ja_JP.UTF8"
sh -c "sleep 5; exec proot-distro login ubuntu --user termux --shared-x11 --env DISPLAY=\$DISPLAY --env DBUS_SESSION_BUS_ADDRESS=\$DBUS_SESSION_BUS_ADDRESS -- env LANG=\${lang} fcitx5" &
```

**Reply by jerrychan7** [reply]
Does your fcitx5 have a candidate box? My fcitx5, installed in termux (not Ubuntu in proof), doesn't have a candidate box when typing. I believe the same issue likely exists when typing complex kanji in Japanese.

**Reply by shiumano** [reply]
Are the settings configured correctly? Please check using `fcitx5-diagnose`.

**Reply by jerrychan7** [reply]
I checked—I wasted several days on it with no results. I suspect the fcitx5 in Termux Packages is missing some critical runtime components. Anyway, I have ibus running properly now.
